flink-streaming-java/src/test/java/org/apache/flink/streaming/api/transformations/GetTransitivePredecessorsTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.transformations;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.dag.Transformation;
+import org.apache.flink.api.java.typeutils.GenericTypeInfo;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.api.operators.TwoInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for {@code getTransitivePredecessors} method of {@link Transformation}. */
+class GetTransitivePredecessorsTest extends TestLogger {
+
+    private Transformation<Integer> commonNode;
+    private Transformation<Integer> midNode;
+
+    @BeforeEach
+    void setup() {
+        commonNode = new TestTransformation<>("commonNode", new MockIntegerTypeInfo(), 1);
+        midNode =
+                new OneInputTransformation<>(
+                        commonNode,
+                        "midNode",
+                        new DummyOneInputOperator(),
+                        new MockIntegerTypeInfo(),
+                        1);
+    }
+
+    @Test
+    void testTwoInputTransformation() {
+        Transformation<Integer> topNode =
+                new TwoInputTransformation<>(
+                        commonNode,
+                        midNode,
+                        "topNode",
+                        new DummyTwoInputOperator<>(),
+                        midNode.getOutputType(),
+                        1);
+        List<Transformation<?>> predecessors = topNode.getTransitivePredecessors();
+        assertThat(predecessors.size()).isEqualTo(3);
+        assertThat(((TestTransformation<?>) commonNode).getNumGetTransitivePredecessor())
```

Comment:
We may change the definition of `commonNode` as `TestTransformation<?>` as that we do not need to conduct type casting each time.